src/Spryker/Zed/AppPayment/Business/Payment/Method/PaymentMethod.php

```diff
@@ -0,0 +1,295 @@
+        protected AppPaymentConfig $appPaymentConfig,
+        protected PaymentMethodMessageSender $paymentMethodMessageSender,
+        protected AppPaymentRepositoryInterface $appPaymentRepository
+    ) {
+    }
+
+    public function configurePaymentMethods(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
+    {
+        // Do not send the message(s) when App is in state "disconnected" or when the app is marked as inactive.
+        if ($appConfigTransfer->getStatus() === AppKernelConfig::APP_STATUS_DISCONNECTED || $appConfigTransfer->getIsActive() === false || !($this->appPaymentPlatformPlugin instanceof AppPaymentPlatformPaymentMethodsPluginInterface)) {
+            return $appConfigTransfer;
+        }
+
+        $tenantIdentifier = $appConfigTransfer->getTenantIdentifierOrFail();
+
+        $paymentMethodConfigurationRequestTransfer = new PaymentMethodConfigurationRequestTransfer();
+        $paymentMethodConfigurationRequestTransfer->setAppConfig($appConfigTransfer);
+
+        $paymentMethodCollectionResponseTransfer = $this->appPaymentPlatformPlugin->configurePaymentMethods($paymentMethodConfigurationRequestTransfer);
+
+        // Get current persisted Tenants payment methods
+        $persistedPaymentMethodTransfers = $this->appPaymentRepository->getTenantPaymentMethods($tenantIdentifier);
+
+        // Get current configured payment methods
+        /** @var array<\Generated\Shared\Transfer\PaymentMethodTransfer> $configuredPaymentMethodTransfers */
+        $configuredPaymentMethodTransfers = $paymentMethodCollectionResponseTransfer->getPaymentMethods()->getArrayCopy();
+
+        $paymentMethodTransfersToAdd = $this->getPaymentMethodsToAdd($persistedPaymentMethodTransfers, $configuredPaymentMethodTransfers);
+        $paymentMethodTransfersToUpdate = $this->getPaymentMethodsToUpdate($persistedPaymentMethodTransfers, $configuredPaymentMethodTransfers);
+        $paymentMethodTransfersToDelete = $this->getPaymentMethodsToDelete($persistedPaymentMethodTransfers, $configuredPaymentMethodTransfers);
+
+        foreach ($paymentMethodTransfersToAdd as $paymentMethodTransferToAdd) {
+            $this->addPaymentMethod($paymentMethodTransferToAdd, $appConfigTransfer);
+        }
+
+        foreach ($paymentMethodTransfersToUpdate as $paymentMethodTransferToUpdate) {
+            $this->updatePaymentMethod($paymentMethodTransferToUpdate, $appConfigTransfer);
+        }
+
+        foreach ($paymentMethodTransfersToDelete as $paymentMethodTransferToDelete) {
+            $this->deletePaymentMethod($paymentMethodTransferToDelete, $appConfigTransfer);
+        }
+
+        return $appConfigTransfer;
+    }
+
+    protected function addPaymentMethod(PaymentMethodTransfer $paymentMethodTransfer, AppConfigTransfer $appConfigTransfer): void
+    {
+        // Add the passed configuration to the default configuration.
+        $paymentMethodAppConfigurationTransfer = $this->getDefaultPaymentMethodAppConfiguration();
```

Comment:
it looks "too hidden" from the app developer. 
so a developer even don't understand that default API URLs (/private) will be sent to SCOS, even no ability to define/overwrite them from the plaform plugin.

What if the app developer should use this method explicitly inside the plugin and have the ability to overwrite URLs?
e.g. via `AppPaymentConfig->getDefaultPaymentMethodAppConfiguration()`.

Reply:
Let's implement it when it gets requested. The point of this is that these packages provide these endpoints so a developer does not need to implement or even know them.

Documentation will help here. I added a more precise one for devs to the README.md

Reply:
My main concern is as an app developer I cannot change URLs... Maybe it won't be used, but it makes me think that everything is hardcoded in the package and to change it the only way is overwrite methods :)

Reply:
We are the developers and if we see the need for being able to change we can implement.